sterror/sterror.go

```diff
@@ -0,0 +1,157 @@
+package sterror
+
+import (
+	"errors"
+)
+
+type Op string
+
+func (o Op) String() string {
+	return string(o)
+}
```

Comment:
Are the String functions nessesary? Maybe for %s statements. But I'd try without them first.